exoplanet-transit-simulator/src/LightcurveView.jsx

```diff
@@ -123,27 +123,44 @@ export default class LightcurveView extends React.Component {
         }
         return a;
     }
-    randomDataSet() {
-        if (!this.props.showSimulatedMeasurements) {
-            return [];
-        }
-        return Array.from(Array(this.props.simMeasurementNumber)).map(
-            () => [Math.random() * 200, Math.random() + 0.5]);
-    }
     /**
      * Generate noise around the lightcurve data.
      */
     getNoiseData(lightcurveData) {
+        const self = this;
```

Comment:
Not a critique but a question for my own understanding, why the assignment of `this` to `self`?

Reply:
Because I need to refer to `this` later on in this function.. inside a nested anonymous function on line 159 (`self.props.noise`).

I usually use `me` for doing this kind of thing, but then I saw Susan use `self` so I'm doing the same thing because it's like Python.

If I didn't save `this` in some variable like this, when I referred to `this` in the function below, it would refer to that anonymous function, instead of this React component that I need to get props off of.